src/actions/sources.js

```diff
@@ -114,8 +114,19 @@ export function newSources(sources: Source[]) {
       source => !getSource(getState(), source.id)
     );
 
+    if (filteredSources.length == 0) {
+      return;
+    }
+
+    dispatch({
+      type: "ADD_SOURCES",
+      sources: filteredSources
+    });
+
     for (const source of filteredSources) {
-      dispatch(newSource(source));
+      dispatch(loadSourceMap(source));
+      dispatch(checkSelectedSource(source));
+      dispatch(checkPendingBreakpoints(source.id));
```

Comment:
oops - i think `dispatch(newSource(source));` was a merge conflict...

we should probably remove `newSource` soon if we don't intend on using it.